File: src/main/java/com/example/kotrip/dto/daytrip/NaverRequestDto.java
package com.example.kotrip.dto.daytrip;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.LocalDate;
import java.util.List;

@Getter
public class NaverRequestDto {
    @JsonProperty("title")
    private String title;

    @JsonProperty("kotrip")
    private List<NaverKotripDto> kotrip;

    @JsonProperty("areaId")
    private int areaId;

    // 생성자, getter 등 필요한 메소드 추가
    public NaverRequestDto(String title, List<NaverKotripDto> kotrip, int areaId) {
        this.title = title;
        this.kotrip = kotrip;
        this.areaId = areaId;
    }
}

File: src/main/kotlin/com/example/kotrip/dto/request/NaverRequestDto.kt
package com.example.kotrip.naver

import com.example.kotrip.dto.Node
import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Getter
import java.time.LocalDate

@Getter
data class NaverRequestDto( // 1개의 여행 스케줄
    @JsonProperty("title") val title: String,
    @JsonProperty("kotrip") val kotrip: List<NaverKotripDto>,
    @JsonProperty("areaId") val areaId: Int
)

@Getter
data class NaverKotripDto( // 여행 1일차, 2일차..
    val date:LocalDate,
    val nodes: List<Node>
)